src/VecSim/algorithms/hnsw/serialization.cpp

```diff
@@ -220,7 +220,8 @@ void HNSWIndexSerializer::saveIndex(const std::string &location) {
     output.close();
 }
 
-void HNSWIndexSerializer::loadIndex(const std::string &location, SpaceInterface<float> *s) {
+void HNSWIndexSerializer::loadIndex(const std::string &location,
+                                    Spaces::dist_func_t<float> dist_func, size_t dim) {
```

Comment:
Isn't the `dim` can be retrieved from the HNSW index that created the sterilizer?

Reply:
I thuoght that the idea is to create an "empty" index, index with some arbitrary dim and function, and then modify it according to the loaded index